mewarpx/mewarpx/diags_store/flux_diagnostic.py

```diff
@@ -0,0 +1,931 @@
+                        ts_dict[x].get_timeseries_by_key(typekey)
+                    )
+                    for (label, x) in
+                    zip(label_list, list(ts_dict.keys()))
+                ],
+                ax=ax,
+                xlabel=xlabel,
+                ylabel=qtydict['ylabel'],
+                title=qtydict['title'],
+                titlesize=18,
+                labelsize=16,
+                alpha=0.7,
+                legend=False
+            )
+
+        fig.legend(ax.get_lines(), label_list, 'lower center', fontsize=16,
+                   frameon=True, ncol=3)
+        fig.tight_layout()
+        fig.subplots_adjust(bottom=0.13 + int((len(label_list)-1) / 3) * 0.04,
+                            hspace=0.33)
+
+        if save:
+            fig.savefig(
+                os.path.join(
+                    self.write_dir, 'flux_plots_{:010d}.png'.format(mwxrun.get_it())
+                ), dpi=300
+            )
+            plt.close(fig)
+
+        return fig
+
+    def get_net_flux_timeseries(self, electrode_name, flux_type='J'):
+        """Sum the emitted and absorbed flux across all species for a given
+        electrode, and return a timeseries array containing the net flux from
+        the electrode surface into the simulation volume.
+
+        Arguments:
+            electrode_name (str): Surface name such as 'cathode' or 'anode'.
+                Must be present in the RunInfo injector_dict and surface_dict in
+                order to be a valid key.
+            flux_type (str): Either 'J', 'P', 'dQ', or 'n'. Defaults to 'J'.
+
+        Returns:
+            flux_timeseries (np.ndarray): n x 2 array containing the time in
+                seconds in first column and the net flux from the electrode in
+                second column.
+        """
+        flux_data = timeseries.concat_crop_timeseries([
+            val for key, val in self.fullhist_dict.items()
+            if electrode_name in key
+        ])
+
+        if flux_data is None:
+            raise ValueError(
+                'No flux data found for "%s" electrode!' % electrode_name
+            )
+
+        return flux_data.get_timeseries_by_key(flux_type)
+
+
+class FluxDiagnostic(FluxDiagBase):
+
+    """Handles writing out charge injection and absorption."""
+
+    def __init__(self, diag_steps, scraper, runinfo,
+                 write_dir='diags/fluxes', overwrite=True,
+                 history_maxlen=5000, sig_figs=6,
+                 printed_qtys=None,
+                 check_charge_conservation=True,
+                 print_per_diagnostic=True, print_total=False,
+                 plot=True, profile_decorator=None, **kwargs):
+        """Generate and install function to write out fluxes.
+
+        Arguments:
+            diag_steps (int): Number of steps between each output
+            scraper (warp.ParticleScraper): The particle scraper is instructed
+                to save E_total for lost particles.
```

Comment:
We can remove the scraper from the argument list. We won't need it (or even have one) in WarpX.